index.html

```diff
@@ -414,41 +418,59 @@ <h2>
           <a>installability signals</a> to warrant <a>installation</a> of the
           web application.
           </li>
+          <li>The installation process can occur through a <dfn>site-triggered
+          install prompt</dfn>: the site can programmatically request that the
+          user agent present an install prompt to the user. The user agent MAY
+          restrict the availability of this feature to cases where, for
+          instance, there are sufficient <a>installability signals</a> to
+          warrant <a>installation</a> of the web application.
+          </li>
         </ul>
+        <p>
+          In any case, the user agent MUST NOT <a>present an install prompt</a>
+          if the document is not <a>installable</a>.
+        </p>
         <p>
           Prior to presenting an <a>automated install prompt</a>, a user agent
           MUST run the <a>steps to notify that an install prompt is
           available</a>, to give the site the opportunity to prevent the
           default action (which is to install the application). Alternatively,
-          the user agent MAY run the <a>steps to notify that an install prompt
-          is available</a> at any time, giving the site the opportunity to show
-          a <a>site-triggered install prompt</a> without automatically showing
-          the prompt.
+          the user agent MAY, at any time (only if the document is
+          <a>installable</a>), run the <a>steps to notify that an install
+          prompt is available</a> at any time, giving the site the opportunity
+          to show a <a>site-triggered install prompt</a> without automatically
+          showing the prompt.
         </p>
         <p>
-          In either case, when a user agent <dfn data-lt=
-          "present an install prompt|presenting an install prompt">presents an
-          install prompt</dfn>, the end-user's choice is represented either
-          "accepted" or "dismissed". These values are represented in the API of
-          this specification via the <a>AppBannerPromptOutcome</a> enum.
+          To <dfn data-lt="presenting an install prompt">present an install
+          prompt</dfn>:
         </p>
+        <ol>
+          <li>Show some user-agent-specific UI, asking the user whether to
+          proceed with installing the app. See <a href=
+          "#installation-sec">privacy and security considerations</a> for
+          recommendations relating to this UI. The <var>result</var> of this
+          choice is either <a data-link-for=
+          "AppBannerPromptOutcome">accepted</a> or <a data-link-for=
+          "AppBannerPromptOutcome">dismissed</a>.
+          </li>
+          <li>If <var>result</var> is <a data-link-for=
+          "AppBannerPromptOutcome">accepted</a>, then in parallel, run the <a>
+            steps to install the web application</a>.
+          </li>
+          <li>Return <var>result</var>.
```

Comment:
This is ambiguous: is `result` returned before "in parallel" runs, or after? As it reads (to me), it's returning before, which I think is incorrect, right?

Reply:
It's intended to be before (i.e., your reading is correct). I don't think it's ambiguous: that's precisely the meaning of "in parallel" (otherwise I wouldn't have said "in parallel").

The result of "present an install prompt" should be the user's choice, not the success of the installation. The result of this promise reflects the user's choice, whereas the `appinstalled` event reflects whether it was actually installed.

At least, that was my intention. Now I'm not so sure. Looking at Chrome's [app_banner_manager](https://cs.chromium.org/chromium/src/chrome/browser/banners/app_banner_manager_desktop.cc?l=197), the code that sends BannerAccepted (which results in the promise resolving with `"accepted"`) has a comment:

```C++
  // Catch only kSuccessNewInstall and kUserInstallDeclined. Report nothing on
  // all other errors.
```

implying that maybe we resolve with `"accepted"` if the user accepts AND installation is success, and perhaps we never resole if the user accepts and installation fails??? @dominickng Can you confirm what Chrome is doing here?

We should decide what the (currently unspecified) behaviour should be if the user accepts the banner but the installation fails: if we want to resolve with `"accepted"` then the proposed text is correct. If not, we should delete "in parallel" and figure out what to resolve with (perhaps we should introduce a new `"accepted-but-failed"` result?

Reply:
Discussed with @dominickng. Chrome does:

- On Android, it resolves with `"accepted"` as soon as the user chooses to install, regardless of whether the installation actually succeeds (matching the proposed "in parallel" text).
- On Windows / Chrome OS / Linux / macOS, if the user accepts and the install "fails", the promise won't resolve. But the failures are all very much edge cases (like something has gone horribly wrong, or the web contents is destroyed, etc), not expected to happen in normal situations. We would consider these edge cases Chrome bugs if they happened. For all intents and purposes, the proposed "in parallel" text matches our behaviour.

So I would prefer to keep this as-is.

Reply:
Ok, alternative:

```
<li>Return |result| and in parallel: .... 
```
 
That way it's clear that in parallel goes off to do other things independently of |result|.

Reply:
OK that works.